src/frontEnd/ProjectExplorer.py

```diff
@@ -226,51 +230,101 @@ def refreshProject(self):
-        if ok and newBaseFileName:
-            newBaseFileName = str(newBaseFileName)
-            projectPath, projectFiles = list(
-                self.obj_appconfig.project_explorer.items())[indexItem.row()]
+    def renameProject(self):
+        """
+        This function renames the project present in project explorer area
+        it validates first:
+        
+            - If project names is not empty.
+            - Project name does not contain spaces between them.
+            - Project name is different between what it was earlier.
+            - Project name should not exist.
+
+        And after project name is changed it recreates the project explorer tree.
+        """
+        self.indexItem = self.treewidget.currentIndex()
+        self.baseFileName = str(self.indexItem.data())
+        self.newBaseFileName, ok = QtGui.QInputDialog.getText(self, 'Rename Project', 'Project Name:',
+                                                            QtGui.QLineEdit.Normal, self.baseFileName)
+        if ok and self.newBaseFileName:
+            print("=================")
+            print(self.newBaseFileName)
+            print("=================")
+            self.newBaseFileName = str(self.newBaseFileName)
+            projectPath, projectFiles = list(self.obj_appconfig.project_explorer.items())[self.indexItem.row()]
             updatedProjectFiles = []
 
-            # rename files matching project name
-            for projectFile in projectFiles:
-                if baseFileName in projectFile:
-                    oldFilePath = os.path.join(projectPath, projectFile)
-                    projectFile = projectFile.replace(
-                        baseFileName, newBaseFileName, 1)
-                    newFilePath = os.path.join(projectPath, projectFile)
-                    print ("Renaming " + oldFilePath + " to " + newFilePath)
-                    os.rename(oldFilePath, newFilePath)
-
-                    updatedProjectFiles.append(projectFile)
-
-            # rename project folder
-            updatedProjectPath = newBaseFileName.join(
-                projectPath.rsplit(baseFileName, 1))
-            print ("Renaming " + projectPath + " to " + updatedProjectPath)
-            os.rename(projectPath, updatedProjectPath)
-
-            # update project_explorer dictionary
-            del self.obj_appconfig.project_explorer[projectPath]
-            self.obj_appconfig.project_explorer[updatedProjectPath] = (
-                updatedProjectFiles
-            )
+            self.workspace = self.obj_appconfig.default_workspace['workspace']
+            self.newBaseFileName = str(self.newBaseFileName).rstrip().lstrip()
+            self.projDir = os.path.join(self.workspace, str(self.newBaseFileName))
 
-            # save project_explorer dictionary on disk
-            json.dump(self.obj_appconfig.project_explorer,
-                      open(self.obj_appconfig.dictPath, 'w'))
+            if self.newBaseFileName == "":
```

Comment:
```suggestion
            if newBaseFileName == "":
```
don't use self. for local variables